Tests/test_image_filter.py

```diff
@@ -5,90 +5,110 @@
 from .helper import assert_image_equal, hopper
 
 
-def test_sanity():
-    def apply_filter(filter_to_apply):
-        for mode in ["L", "RGB", "CMYK"]:
-            im = hopper(mode)
-            out = im.filter(filter_to_apply)
-            assert out.mode == im.mode
-            assert out.size == im.size
-
-    apply_filter(ImageFilter.BLUR)
-    apply_filter(ImageFilter.CONTOUR)
-    apply_filter(ImageFilter.DETAIL)
-    apply_filter(ImageFilter.EDGE_ENHANCE)
-    apply_filter(ImageFilter.EDGE_ENHANCE_MORE)
-    apply_filter(ImageFilter.EMBOSS)
-    apply_filter(ImageFilter.FIND_EDGES)
-    apply_filter(ImageFilter.SMOOTH)
-    apply_filter(ImageFilter.SMOOTH_MORE)
-    apply_filter(ImageFilter.SHARPEN)
-    apply_filter(ImageFilter.MaxFilter)
-    apply_filter(ImageFilter.MedianFilter)
-    apply_filter(ImageFilter.MinFilter)
-    apply_filter(ImageFilter.ModeFilter)
-    apply_filter(ImageFilter.GaussianBlur)
-    apply_filter(ImageFilter.GaussianBlur(5))
-    apply_filter(ImageFilter.BoxBlur(5))
-    apply_filter(ImageFilter.UnsharpMask)
-    apply_filter(ImageFilter.UnsharpMask(10))
-
+@pytest.mark.parametrize(
+    "filter_to_apply",
+    (
+        ImageFilter.BLUR,
+        ImageFilter.CONTOUR,
+        ImageFilter.DETAIL,
+        ImageFilter.EDGE_ENHANCE,
+        ImageFilter.EDGE_ENHANCE_MORE,
+        ImageFilter.EMBOSS,
+        ImageFilter.FIND_EDGES,
+        ImageFilter.SMOOTH,
+        ImageFilter.SMOOTH_MORE,
+        ImageFilter.SHARPEN,
+        ImageFilter.MaxFilter,
+        ImageFilter.MedianFilter,
+        ImageFilter.MinFilter,
+        ImageFilter.ModeFilter,
+        ImageFilter.GaussianBlur,
+        ImageFilter.GaussianBlur(5),
+        ImageFilter.BoxBlur(5),
+        ImageFilter.UnsharpMask,
+        ImageFilter.UnsharpMask(10),
+    ),
+)
+@pytest.mark.parametrize("mode", ("L", "RGB", "CMYK"))
+def test_sanity(filter_to_apply, mode):
+    im = hopper(mode)
+    out = im.filter(filter_to_apply)
+    assert out.mode == im.mode
+    assert out.size == im.size
+
+
+@pytest.mark.parametrize("mode", ("L", "RGB", "CMYK"))
+def test_sanity_error(mode):
     with pytest.raises(TypeError):
-        apply_filter("hello")
-
-
-def test_crash():
-
-    # crashes on small images
-    im = Image.new("RGB", (1, 1))
-    im.filter(ImageFilter.SMOOTH)
+        im = hopper(mode)
+        out = im.filter("hello")
+        assert out.mode == im.mode
+        assert out.size == im.size
```

Comment:
These two asserts are not needed.

Reply:
resolved with 826ab4b17c1c4622a7210b4c72d50606ed2b2d2a